Common/src/gRPC/Services/Submitter.cs

```diff
@@ -337,6 +335,59 @@ await payloadUploadTasks.WhenAll()
                                               cancellationToken)
                                    .ConfigureAwait(false);
 
+    if (result.Status != ResultStatus.Completed)
+    {
+
+      var taskData = await taskTable_.ReadTaskAsync(result.OwnerTaskId,
+                                                    cancellationToken)
+                                     .ConfigureAwait(false);
+
+      switch (taskData.Status)
+      {
+        case TaskStatus.Processed:
+        case TaskStatus.Completed:
+          break;
+        case TaskStatus.Error:
+        case TaskStatus.Failed:
+          await responseStream.WriteAsync(new ResultReply
+                                          {
+                                            Error = new TaskError
+                                                    {
+                                                      TaskId = taskData.TaskId,
+                                                      Error =
+                                                      {
+                                                        new Error
+                                                        {
+                                                          Detail     = taskData.Output.Error,
+                                                          TaskStatus = taskData.Status,
+                                                        },
+                                                      },
+                                                    },
+                                          },
+                                          CancellationToken.None)
+                              .ConfigureAwait(false);
+          return;
+        case TaskStatus.Creating:
+        case TaskStatus.Timeout:
+        case TaskStatus.Canceling:
+        case TaskStatus.Canceled:
+        case TaskStatus.Submitted:
+        case TaskStatus.Dispatched:
+        case TaskStatus.Processing:
+          await responseStream.WriteAsync(new ResultReply
+                                          {
+                                            NotCompletedTask = taskData.TaskId,
+                                          },
+                                          CancellationToken.None)
+                              .ConfigureAwait(false);
+          return;
+
+        case TaskStatus.Unspecified:
+        default:
+          throw new ArgumentOutOfRangeException();
```

Comment:
Pourquoi mettre un out of range? Cela ne pourrait pas etre plus precis du style "Status does not exist" (au moins au niveau du log ou mets tu cela uniquement parceque un default est obligatoire?

Reply:
Le outofrange, c'est le truc par defaut. Et il s'avere qu'un defaut est obligatoire. On utilise l'option de resharper qui detecte que si le switch case fait tous les cas mais il faut un defaut. Si on rentre dans ce cas, il y a vraiment un problème